### ISSUE TITLE
Task list: Shipping smart defaults

### ISSUE THREAD

**Issue by verofasulo** [woothemes/woocommerce#32318]
**User story:**
As a new merchant
I want to focus on the most important setup tasks
So I can review some details later

**Description:**
With Jetpack installed and connected, alongside WCS, we have the opportunity to set things up on the users' behalf, and ask them to review and confirm, rather than start from scratch and think.

**Design overview (with new spotlight component):**
cJQeyyhx9PraQrY3YAtCGM-fi-705%3A80783

**Acceptance Criteria:**

- [ ] when the defaults are automatically turned on, the shipping options are the following: 

- store country: free shipping 
- other countries: flat rate

- [ ] If the store sells physical products, has JP and WCS installed and connected, and is located in the US: 

- Automatically turn on automated shipping options. 
- Move the "Review shipping options task" to the "Things to do next" section.
- Address users to WooCommerce > Settings > Shipping once clicked on the task on the home page.
- Display intro tooltips in Shipping Settings: Shipping zones, shipping methods, WCS tab.

![US_Canada_Australia - Plugins Installed](https://user-images.githubusercontent.com/98944206/159504026-16f514c4-4c39-4b87-aeac-3a91f4c91628.gif)

- [ ] If the store sells physical products, and is not located in US/Canada/Australia/UK (irrelevant if JP is installed or not): 

- Automatically turn on automated shipping options. 
- Move the "Review shipping options task" to the "Things to do next" section.
- Address users to WooCommerce > Settings > Shipping once clicked on the task on the home page.
- Display intro tooltips in Shipping Settings: Shipping zones and shipping methods.

![No US_Canada_Australia](https://user-images.githubusercontent.com/98944206/159504071-5660c34f-6b33-4902-992a-28a55a8e27af.gif)

- [ ] If the store sells physical products and is located in US, but JP and WCS are not installed: 

- Automatically turn on automated shipping options. 
- Move the "Review shipping options task" to the "Things to do next" section.
- Address users to WooCommerce > Settings > Shipping once clicked on the task on the home page.
- Display intro tooltips in Shipping Settings: Shipping zones, shipping methods, WCS as recommended extension to be installed. 
- Implement the new flow to install and connect WCS.
- This new flow requires embedding an iFrame to avoid sending users to the external Jetpack page. It's feasible on all browsers apart from Safari; we'll keep the current flow (external Jetpack page) for this specific browser.

![US_Canada_Australia - Plugin not installed (1)](https://user-images.githubusercontent.com/98944206/159506263-a846740e-b536-4c7f-896f-398d4c2aa238.gif)

- [ ] If we don't know if the store sells physical products and where it's located, and JP is not installed:

- Display the "Set up shipping" task in the "Get ready to start selling" section.
- Address users to a single page with the following steps: Set your store location, Review your shipping options, Enable shipping label printing and discounted rates (only if the store location added in the first step is US, Canada or Australia), Connect your store
- Connect your store step: embed an iFrame to keep the user in the flow.

![No Location (1)](https://user-images.githubusercontent.com/98944206/159508460-b20ac838-d517-4679-8e6d-60f3f97bb2f6.gif)

- [ ] If the store sells digital products:

- Do not display the shipping task on the home page. 

- [ ] If the store is located in the UK, Australia or Canada:

- Display the "Set up shipping" task in the "Get ready to start selling" section.
- Same flow as now with a few optimizations in terms of copy and visual.

https://user-images.githubusercontent.com/98944206/161233363-b71277bc-82f1-48ef-bb25-151b7bc4ac09.mov

**Event tracking:**
To be assessed by the team - some things I'd like to measure:

- Task completion rate
- After how many tasks users click on the Review shipping options task
- % of stores where the first order is placed without reviewing the shipping options

**Comment by pmcpinto** [comment]
I'll add this to the Cycle 23 report for effort estimation

**Comment by pmcpinto** [comment]
@becdetat can you check if one of the teams can estimate the effort required to implement this? The smart default is something that also needs to be implemented for woocom. Thanks!

**Comment by becdetat** [comment]
@mattsherman Can Mothra estimate this please? We have the tooltips implemented but not the step counter or overlays. Possibly building that component out should be estimated separately and as another ticket.

**Comment by adrianduffell** [comment]
@verofasulo Thanks for putting this issue together. Just a quick question hopefully 😄 

What is meant by `Automatically turn on automated shipping options` in the acceptance criteria?

**Comment by verofasulo** [comment]
Hey @adrianduffell 😄 

It means that when we enable the shipping smart defaults, the shipping options are the following:

- store country: free shipping (e.g. if the store is based in Lisbon, the store country is Portugal)

- other countries: flat rate (e.g. all the other countries apart from Portugal). The flat rate should be automatically defined based on the average shipping cost. It's a bit difficult to define the right average cost now as the level of granularity is very high (the average cost varies considerably if the store ships to Europe or to the USA, for example), but we can start with an approximation and then track how many users tweak the value. 

Let me know if I answered your question 🤗

**Comment by verofasulo** [comment]
Following up on the [new spotlight component initiative](https://ghidorahp2.wordpress.com/2022/04/13/shared-in-app-spotlight-component/), please implement the new spotlight that’s part of a larger design element used for help and guidance.

The design will be the same proposed by Dotcom with a few tweaks to make it consistent with WooCommerce styles and usage.

![image](https://user-images.githubusercontent.com/98944206/164425205-7e0524ca-35be-4ec1-a2e8-69c665805acb.png)

**Design overview:** 
cJQeyyhx9PraQrY3YAtCGM-fi-888%3A112694

**Differences:**
- Use WooCommerce button styles and colors.
- Remove the Minimize icon.
- Use SF Pro Text for the title in place of Inter.

**Updated shipping smart defaults designs (with the new spotlight component)**: 
cJQeyyhx9PraQrY3YAtCGM-fi-705%3A80783

**Comment by chihsuan** [comment]
> Possibly building that component out should be estimated separately and as another ticket.

I created another issue for spotlight component [#32892](https://github.com/woocommerce/woocommerce/issues/32892).

**Comment by chihsuan** [comment]
> Implement the new flow to install and connect WCS.

> Address users to a single page with the following steps: Set your store location, Review your shipping options, Enable shipping label printing and discounted rates (only if the store location added in the first step is US, Canada or Australia), Connect your store

So besides having `automated shipping options`, we'll need to create a new flow and optimize the current flow. I think we could create sub-tasks for them before we start to work on this.

**Comment by chihsuan** [comment]
@verofasulo Thanks for providing a details task description! 😄 

I have two questions:

>  If the store sells digital products:

> Do not display the shipping task on the home page.

![Screen Shot 2022-05-06 at 15 32 52](https://user-images.githubusercontent.com/4344253/167087697-e617f6cd-5731-4641-b1f6-ab25905b2558.png)


- If a user selects both downloads (digital) and physical products -> use physical products rules.
- if a user selects the other product types -> same as  "If we don't know if the store sells physical products and where it's located, and **JP is not installed**"

Is my understanding correct?

And what should we do "If we don't know if the store sells physical products and where it's located, and **JP is installed**."

**Comment by verofasulo** [comment]
Hey @chihsuan 😄

> If a user selects both downloads (digital) and physical products -> use physical products rules.

That's correct.

> if a user selects the other product types -> same as "If we don't know if the store sells physical products and where it's located, and JP is not installed"

If the user selects the other product types (one or more than one) without picking physical products or digital products, we are in the same situation as above. The other product types can be applied both to digital and physical products, so we go with the rules of physical products. 
In the specific case you mentioned, if JP is not installed we follow the use case `"If we don't know if the store sells physical products and where it's located, and JP is not installed"`. if Jetpack is installed, please follow the same use case removing the last step (connect your store) as I guess it's unnecessary 😄

**Comment by becdetat** [comment]
I'm checking on the status of this as it looks like it will be part of cycle 26. Do we have enough information to proceed (@chihsuan  @adrianduffell )? It's also rather large and has a number of cases, can we break it down into smaller pieces? This would make it easier to manage and might make it possible for a couple of people to work on this in parallel.

**Comment by chihsuan** [comment]
> Do we have enough information to proceed? 

@becdetat Yup, I believe this is ready to proceed. Actually, we estimated it in Cycle 25 Planning call.

>  It's also rather large and has a number of cases, can we break it down into smaller pieces?

Absolutely, I'll see if I can do it as a porter this week.

**Comment by chihsuan** [comment]
> It could be challenging to have this wrapped under ExPlat, we should discuss with the Product team to see if we really need to experiment -  ([notes](https://docs.google.com/document/d/1r1NuprkOmFgXVzFDwshCa-Qu-qXoZgiplxnqrp4sFJs/edit#))

@pmcpinto @verofasulo We discussed this in our Cycle 25 Planning call. If I remember correctly,  because we will add/change many things in this task, it might not be appropriate to perform A/B tests.  cc @ilyasfoo

**Comment by pmcpinto** [comment]
👍 Sounds good to me. As Veronica included in the description, our goal is to start testing this approach in some countries.

**Comment by ilyasfoo** [comment]
> If I remember correctly, because we will add/change many things in this task, it might not be appropriate to perform A/B tests. 

@chihsuan Thanks for raising that, I agree that the change is too much for us to be able to meaningfully measure what contributes to any improvements/degradation. Makes sense to not use experiments for this.

@pmcpinto Without explat, we won't be able to disable the feature remotely (such as disabling an experiment in Abacus) unless we build a remote-feature-toggling mechanism. Do you think it's okay to not have that?

**Comment by adrianduffell** [comment]
> Absolutely, I'll see if I can do it as a porter this week.

@chihsuan Thanks! I'll have a couple of internal P2 project threads for this at end of the week as well, where we can group the new issues.

I think there are two main projects here:
* Smart defaults for Shipping
* New Jetpack Connection Flow

**Comment by chihsuan** [comment]
> I think there are two main projects here:

> Smart defaults for Shipping
> New Jetpack Connection Flow

@adrianduffell Sounds good! I think we might only have one task for "New Jetpack Connection Flow", but it's true that it's not really related to Shipping.

**Comment by chihsuan** [comment]
I drew a rough diagram to help us understand AC better.

![Screen Shot 2022-06-09 at 16 43 02](https://user-images.githubusercontent.com/4344253/172804730-e254552c-b52f-4296-ad84-80c86d2ae19a.png)

**Comment by chihsuan** [comment]
I've split this into the following issues:
- https://github.com/woocommerce/woocommerce/issues/33363
- https://github.com/woocommerce/woocommerce/issues/33364
- https://github.com/woocommerce/woocommerce/issues/33365
- https://github.com/woocommerce/woocommerce/issues/33366
- https://github.com/woocommerce/woocommerce/issues/33367
- https://github.com/woocommerce/woocommerce/issues/33368
- https://github.com/woocommerce/woocommerce/issues/33369
- https://github.com/woocommerce/woocommerce/issues/33370
- ~https://github.com/woocommerce/woocommerce/issues/33371~

**Comment by pmcpinto** [comment]
> @pmcpinto Without explat, we won't be able to disable the feature remotely (such as disabling an experiment in Abacus) unless we build a remote-feature-toggling mechanism. Do you think it's okay to not have that?

Sorry, I missed this ping. I think that's ok, and it's mainly the reason why we want to start first with just US stores. We could also have a 99%/1% a/b test just to be able to disable it remotely,

**Comment by SamratBiswas1** [comment]
I was talking to @pmcpinto and he mentioned about whether we should trigger shipping task completion at default settings or should we do it after the user has reviewed shipping options. Since the user can accept orders and ship at the default setting; from user standpoint their store has shipping capability and have completed the task. Hence, I am of the opinion that we should trigger shipping task complete at the default setting.

**Comment by pmcpinto** [comment]
👍 I think that suggestion makes sense.

@chihsuan is it easy to change the `wcadmin_tasklist_task_completed` `task_name` : `shipping ` behavior in this particular case?

**Comment by chihsuan** [comment]
> is it easy to change the wcadmin_tasklist_task_completed task_name : shipping behavior in this particular case?

@pmcpinto Yes! So we want to change to trigger the `wcadmin_tasklist_task_completed` `task_name `: `shipping` event at default settings but still mark the task as completed when the user review shipping options, right?

**Comment by pmcpinto** [comment]
Correct. When do you think we should fire wcadmin_tasklist_task_completed task_name : shipping in this case? When the user visits the home screen?!

**Comment by chihsuan** [comment]
> When do you think we should fire wcadmin_tasklist_task_completed task_name : shipping in this case? When the user visits the home screen?!

Yup, when the user visits the home screen. Currently, we only set the smart default setting when a user visits the home screen.
- #33547

**Comment by pmcpinto** [comment]
👍  makes sense. Thanks for the clarification :)

**Comment by ilyasfoo** [comment]
> When do you think we should fire wcadmin_tasklist_task_completed task_name : shipping in this case? When the user visits the home screen?

@SamratBiswas1 @pmcpinto @chihsuan Actually, this is already an existing behaviour since when we've set up default shipping zones via smart defaults, [the completion](https://github.com/woocommerce/woocommerce/blob/97f3e2d6093d1f3f164fb5b28ad3c3c686999c63/plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/Shipping.php#L71-L73) logic returns true, thus completion track is fired when homescreen is loaded. 🥳

**Comment by chihsuan** [comment]
Closing this since we've completed it in cycle 26.
